pkg/spoke/addon/configuration.go

```diff
@@ -60,13 +62,25 @@ func getAddOnInstallationNamespace(addOn *addonv1alpha1.ManagedClusterAddOn) str
 	return installationNamespace
 }
 
+// getAddonInstallMode returns addon installation mode, mode could be `Default` or `Hosted`
+func getAddonInstallMode(addOn *addonv1alpha1.ManagedClusterAddOn) string {
```

Comment:
return a boolean is sufficient. You could call it isAddonRunOnManagementCluster etc.

Reply:
change it to `isAddonRunningOutsideManagedCluster` since it is more like a description of hosted mode